NWM Dynamic Data Dependency and New NWM 3.0 Services

CoreyKrewson-NOAA opened the pull request:
This PR allows the pipeline to change data dependency based on environment variables. This will allow us to run our pipelines using para data in one environment while using prod data in a different environment. 

This PR also adds the following new products associated with NWM 3.0

**Alaska**
- ana_streamflow_ak
- ana_past_72hr_accum_precip_ak
- mrf_gfs_10day_accum_precip_ak
- mrf_nbm_10day_accum_precip_ak
- mrf_gfs_10day_accum_precip_ak
- srf_15hr_accum_precip_ak

**NBM**
- mrf_nbm_10day_accum_precip
- mrf_nbm_10day_high_water_arrival_time
- mrf_nbm_10day_max_high_flow_magnitude
- mrf_nbm_10day_max_inundation
- mrf_nbm_10day_peak_flow_arrival_time
- mrf_nbm_10day_rapid_onset_flooding

CoreyKrewson-NOAA committed:
setup ana_streamflow_ak

CoreyKrewson-NOAA committed:
setup ana_past_72hr_accum_precip_ak

CoreyKrewson-NOAA committed:
setup mrf_gfs_10day_accum_precip_ak

CoreyKrewson-NOAA committed:
setup mrf_nbm_10day_accum_precip_ak

CoreyKrewson-NOAA committed:
setup mrf_nbm_10day_accum_precip

CoreyKrewson-NOAA committed:
setup srf_15hr_accum_precip_ak

CoreyKrewson-NOAA committed:
setup nbm channel services

CoreyKrewson-NOAA committed:
Merge branch 'ti' into nwm30_transition

CoreyKrewson-NOAA committed:
updated services to use new nwm_dataflow_version environment variable
This will allow us to make different environments be triggered off of different data paths

CoreyKrewson-NOAA committed:
added nwm_dataflow_version to EC2 uris

TylerSchrag-NOAA reviewed (dismissed):
I don't see anything immediately problematic here, and I really like the env variable solution to the prod/para switch!

I know you're thinking about this already, but we'll just have to make sure to add the new para SNS triggers when that's done (and some permissions to access it).

CoreyKrewson-NOAA commented:
Yeah good point Tyler. We may have to implement some type of dynamic
approach for the SNS topics as well so we are only depending on the right
SNS topic for each environment

On Thu, Jul 13, 2023 at 12:11 PM TylerSchrag-NOAA ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> I don't see anything immediately problematic here, and I really like the
> env variable solution to the prod/para switch!
>
> I know you're thinking about this already, but we'll just have to make
> sure to add the new para SNS triggers when that's done (and some
> permissions to access it).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-OWP/hydrovis/pull/493#pullrequestreview-1529093206>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQTVUCKZAFORUNSRFGL66EDXQBB6VANCNFSM6AAAAAA2JK442Y>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
*Corey Krewson*
National Water Center
Geo-Intelligence Division | Office of Water Prediction
NOAA National Weather Service
*Office: (443) 695-0769 *| ***@***.*** ***@***.***>*
*http://water.noaa.gov <http://water.noaa.gov/>*

TylerSchrag-NOAA commented:
Well the good news is that the bazillion initialize pipeline false starts have continued to not be a big deal at all, so adding more false starts from both feeds won't be a big deal. My preference would be to just expand the logic at the start of initialize pipeline to sort that out (which I suppose can go into a pipeline method when the timing is right), instead of messing with dynamic triggers in TF (if that's what you mean).

Initialize pipeline continues to get a bit bloated... but I still really like having all of that in one spot at least.

shawncrawley committed:
Remove atlgulf from configuration

Account for special case of combined coastal custom configuration (i.e. the *_coastal_atlgful and *_coastal_pacific NWM configuration files together comprise the custom-made "coastal" configuration on the Viz processing side.

shawncrawley committed:
rename derived mrf coastal configuration

this changes "medium_range_mem1_coastal" to "medium_range_coastal_mem1"
so that the previous commit to remove "_atlgulf" from the actual NWM
configuration will also work on these custom-derived coastal mrf configs

shawncrawley commented:
I made a few changes, so someone may want to review those (they are very few and small). Otherwise, this looks good to me. I've tested it with the short_range_coastal pipeline, and everything worked. So from my POV, this stuff is ready to be merged and officially deployed to TI for one last round of testing before going to UAT.

shawncrawley reviewed (approved):
Looks good - although the last two commits of changes are my own, so I don't know if @TylerSchrag-NOAA should re-review what's changed since his last review.
